extensions/kubernetes/vanilla/deployment/pom.xml

```diff
@@ -29,10 +29,6 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-kubernetes-client-deployment-internal</artifactId>
         </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-smallrye-metrics-spi</artifactId>
-        </dependency>
```

Comment:
Can you explain this change?

1) I don't get it why it's needed (is this simply left over by mistake?)
2) If this is not left over by mistake, I am pretty sure things will break

Reply:
I think that when I went through and made metrics changes in 1.7 (to remove direct usage of the old metrics SPI in favor of the new Metrics capability checks in core), I forgot to remove this.

Reply:
OK, let's see if CI passes

Reply:
Switch to new API: 
https://github.com/quarkusio/quarkus/blob/master/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java#L103
to retrieve metrics path: 
https://github.com/quarkusio/quarkus/blob/cbe8d21b9b97efdb047e8a1abe8a55104688029a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java#L296

That was all the SPI was used for..